File: CodeBattleJava/src/main/java/ru/codebattle/client/api/BoardElement.java
package ru.codebattle.client.api;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Getter
public enum BoardElement {
    NONE(' '),                    // Пустое место – по которому может двигаться герой

    BRICK('#'),                   // Cтена в которой можно просверлить дырочку слева или справа от героя
                                  // (в зависимости от того, куда он сейчас смотрит)

    PIT_FILL_1('1'),              // Стена со временем зарастает. Когда процес начинается - мы видим таймер
    PIT_FILL_2('2'),
    PIT_FILL_3('3'),
    PIT_FILL_4('4'),

    UNDESTROYABLE_WALL('☼'),      // Неразрушаемая стена - в ней ничего просверлить не получится

    DRILL_PIT('*'),               // В момент сверления мы видим процесс так

    // Боты-охотники
    ENEMY_LADDER('Q'),
    ENEMY_LEFT('«'),
    ENEMY_RIGHT('»'),
    ENEMY_PIPE_LEFT('<'),
    ENEMY_PIPE_RIGHT('>'),
    ENEMY_PIT('X'),

    YELLOW_GOLD('$'),             // Горстка золота жёлтого цвета
    GREEN_GOLD('&'),              // Зелёная горстка золота
    RED_GOLD('@'),                // Золото красного цвета

    // Твой герой в зависимости от того, чем он сейчас занят отображается следующими символами
    HERO_DIE('Ѡ'),                // Герой переживает процесс умирания
    HERO_DRILL_LEFT('Я'),         // Герой сверлит слева от себя
    HERO_DRILL_RIGHT('R'),        // Герой сверлит справа от себя
    HERO_LADDER('Y'),             // Герой находится на лестнице
    HERO_LEFT('◄'),               // Герой бежит влево
    HERO_RIGHT('►'),              // Герой бежит вправо
    HERO_FALL_LEFT(']'),          // Герой падает, смотря влево
    HERO_FALL_RIGHT('['),         // Герой падает, смотря вправо
    HERO_PIPE_LEFT('{'),          // Герой ползёт по трубе влево
    HERO_PIPE_RIGHT('}'),         // Герой ползёт по трубе вправо

    // Тоже твой герой, но под таблеткой тени:
    HERO_SHADOW_DIE('x'),         // Герой-тень переживает процесс умирания
    HERO_SHADOW_DRILL_LEFT('⊰'),  // Герой-тень сверлит слева от себя
    HERO_SHADOW_DRILL_RIGHT('⊱'), // Герой-тень сверлит справа от себя
    HERO_SHADOW_LADDER('⍬'),      // Герой-тень находится на лестнице
    HERO_SHADOW_LEFT('⊲'),        // Герой-тень бежит влево
    HERO_SHADOW_RIGHT('⊳'),       // Герой-тень бежит вправо
    HERO_SHADOW_FALL_LEFT('⊅'),   // Герой-тень падает, смотря влево
    HERO_SHADOW_FALL_RIGHT('⊄'),  // Герой-тень падает, смотря вправо
    HERO_SHADOW_PIPE_LEFT('⋜'),   // Герой-тень ползёт по трубе влево
    HERO_SHADOW_PIPE_RIGHT('⋝'),  // Герой-тень ползёт по трубе вправо

    // Герои других игроков отображаются так
    OTHER_HERO_DIE('Z'),          // Герой переживает процесс умирания
    OTHER_HERO_DRILL_LEFT('⌋'),   // Герой сверлит слева от себя
    OTHER_HERO_DRILL_RIGHT('⌊'),  // Герой сверлит справа от себя
    OTHER_HERO_LADDER('U'),       // Герой находится на лестнице
    OTHER_HERO_LEFT(')'),         // Герой бежит влево
    OTHER_HERO_RIGHT('('),        // Герой бежит вправо
    OTHER_HERO_FALL_LEFT('⊐'),    // Герой падает, смотря влево
    OTHER_HERO_FALL_RIGHT('⊏'),   // Герой падает, смотря вправо
    OTHER_HERO_PIPE_LEFT('Э'),    // Герой ползёт по трубе влево
    OTHER_HERO_PIPE_RIGHT('Є'),   // Герой ползёт по трубе вправо

    // А если герои других игроков под таблеткой тени, то так
    OTHER_HERO_SHADOW_DIE('⋈'),         // Другой герой-тень переживает процесс умирания
    OTHER_HERO_SHADOW_DRILL_LEFT('⋰'),  // Другой герой-тень сверлит слева от себя
    OTHER_HERO_SHADOW_DRILL_RIGHT('⋱'), // Другой герой-тень сверлит справа от себя
    OTHER_HERO_SHADOW_LEFT('⋊'),        // Другой герой-тень находится на лестнице
    OTHER_HERO_SHADOW_RIGHT('⋉'),       // Другой герой-тень бежит влево
    OTHER_HERO_SHADOW_LADDER('⋕'),      // Другой герой-тень бежит вправо
    OTHER_HERO_SHADOW_FALL_LEFT('⋣'),   // Другой герой-тень падает, смотря влево
    OTHER_HERO_SHADOW_FALL_RIGHT('⋢'),  // Другой герой-тень падает, смотря вправо
    OTHER_HERO_SHADOW_PIPE_LEFT('⊣'),   // Другой герой-тень ползёт по трубе влево
    OTHER_HERO_SHADOW_PIPE_RIGHT('⊢'),  // Другой герой-тень ползёт по трубе вправо

    LADDER('H'),              // Лестница - по ней можно перемещаться по уровню
    PIPE('~'),                // Труба - по ней так же можно перемещаться по уровню, но только горизонтально

    PORTAL('⊛'),              // Портал - позволяет телепортироваться на карте в иное место на карте

    SHADOW_PILL('S');         // Таблетка тени - наделяют героя дополнительными способностями

    final char symbol;

    @Override
    public String toString() {
        return String.valueOf(symbol);
    }

    public static BoardElement valueOf(char ch) {
        for (BoardElement el : BoardElement.values()) {
            if (el.symbol == ch) {
                return el;
            }
        }
        throw new IllegalArgumentException("No such element for " + ch);
    }
}

File: CodeBattleNet/Loderunner.Api/BoardElement.cs
/*-
 * #%L
 * Codenjoy - it's a dojo-like platform from developers to developers.
 * %%
 * Copyright (C) 2018 Codenjoy
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * #L%
 */
namespace Loderunner.Api
{
    public enum BoardElement : short
    {
        None = (short)' ',

        Brick = (short)'#',
        PitFill1 = (short)'1',
        PitFill2 = (short)'2',
        PitFill3 = (short)'3',
        PitFill4 = (short)'4',
        UndestroyableWall = (short)'☼',

        DrillPit = (short)'*',

        EnemyLadder = (short)'Q',
        EnemyLeft = (short)'«',
        EnemyRight = (short)'»',
        EnemyPipeLeft = (short)'<',
        EnemyPipeRight = (short)'>',
        EnemyPit = (short)'X',

        YellowGold = (short)'$',
        GreenGold = (short)'&',
        RedGold = (short)'@',

        HeroDie = (short)'Ѡ',
        HeroDrillLeft = (short)'Я',
        HeroDrillRight = (short)'R',
        HeroLadder = (short)'Y',
        HeroLeft = (short)'◄',
        HeroRight = (short)'►',
        HeroFallLeft = (short)']',
        HeroFallRight = (short)'[',
        HeroPipeLeft = (short)'{',
        HeroPipeRight = (short)'}',

        HeroShadowDie = (short)'x',
        HeroShadowDrillLeft = (short)'⊰',
        HeroShadowDrillRight = (short)'⊱',
        HeroShadowLadder = (short)'⍬',
        HeroShadowLeft = (short)'⊲',
        HeroShadowRight = (short)'⊳',
        HeroShadowFallLeft = (short)'⊅',
        HeroShadowFallRight = (short)'⊄',
        HeroShadowPipeLeft = (short)'⋜',
        HeroShadowPipeRight = (short)'⋝',

        OtherHeroDie = (short)'Z',
        OtherHeroDrillLeft = (short)'⌋',
        OtherHeroDrillRight = (short)'⌊',
        OtherHeroLeft = (short)')',
        OtherHeroRight = (short)'(',
        OtherHeroLadder = (short)'U',
        OtherHeroFallLeft = (short)'⊐',
        OtherHeroFallRight = (short)'⊏',
        OtherHeroPipeLeft = (short)'Э',
        OtherHeroPipeRight = (short)'Є',

        OtherHeroShadowDie = (short)'⋈',
        OtherHeroShadowDrillLeft = (short)'⋰',
        OtherHeroShadowDrillRight = (short)'⋱',
        OtherHeroShadowLeft = (short)'⋊',
        OtherHeroShadowRight = (short)'⋉',
        OtherHeroShadowLadder = (short)'⋕',
        OtherHeroShadowFallLeft = (short)'⋣',
        OtherHeroShadowFallRight = (short)'⋢',
        OtherHeroShadowPipeLeft = (short)'⊣',
        OtherHeroShadowPipeRight = (short)'⊢',

        Ladder = (short)'H',
        Pipe = (short)'~',

        Portal = (short)'⊛',

        TheShadowPill = (short)'S'
    }
}
